src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -108,9 +109,15 @@ public Person toModelType() throws IllegalValueException {
 
         final Set<Skill> modelSkills = new HashSet<>(personSkills);
 
-        final String modelMatch = match;
-
-        return new Person(modelName, modelPhone, modelEmail, modelRole, modelSkills, modelMatch);
+        // TODO - Discuss if we should have an overloaded Person method taking in (Person person, String match)
```

Comment:
For reference to what I mean exactly, refer to `matchContactToJob` in matchCommand, `unmatchContactFromJob` in unmatchCommand, `removeJob` in addressBook. There might be other instances that I forgot about.

Reply:
Sorry I don't not fully understand what you mean, can you elaborate a bit more?

Reply:
<img width="706" alt="image" src="https://github.com/user-attachments/assets/1c0b7fbf-e786-4bf6-82ca-730c25f420de">

Basically, the chunk where we construct a new person via calling the getters, I see this in the 3 areas I mentioned. I was thinking of another constructor with parameters (Person person, String match) that will abstract out this duplicated code fragment. 

If we want to remove a match, we simply pass in an empty "" and this overloaded method will automatically make an optional.empty()